tests/test_on_data/test_metadata/test_neuralynx_metadata.py

```diff
@@ -13,56 +10,88 @@
+        assert '"sampling_rate": "2000.0"' in file_metadata["notes"]
+
+        device_metadata = metadata["Ecephys"]["Device"]
+        assert device_metadata[-1] == {"name": "DigitalLynxSX", "description": "Cheetah 5.6.3"}
 
 
 def test_neuralynx_cheetah_v540_metadata():
+    import neo
+    import distutils.version as version
+
     folder_path = NLX_PATH / "Cheetah_v5.4.0" / "original_data"
-    assert get_metadata(str(folder_path)) == dict(
-        session_start_time=datetime.datetime(2001, 1, 1, 0, 0),
-    )
+    metadata = NeuralynxRecordingInterface(folder_path).get_metadata()
+    file_metadata = metadata["NWBFile"]
+
+    assert file_metadata["session_start_time"] == datetime.datetime(2001, 1, 1, 0, 0)
+
+    assert '"recording_closed": "2001-01-01 00:00:00"' in file_metadata["notes"]
+    assert '"ADMaxValue": "32767"' in file_metadata["notes"]
+
+    # the sampling rate and device is only exposed on neo dev as of now. TODO: Remove if after neo >=0.12
+    if version.LooseVersion(neo.__version__) > version.LooseVersion("0.11.0"):
+        assert '"sampling_rate": "1017.375"' in file_metadata["notes"]
+
+        device_metadata = metadata["Ecephys"]["Device"]
+        assert device_metadata[-1] == {"name": "DigitalLynx", "description": "Cheetah 5.4.0"}
 
 
 def test_neuralynx_filtering():
-    file_path = NLX_PATH / "Cheetah_v5.7.4" / "original_data" / "CSC1.ncs"
-    assert (
-        get_filtering(str(file_path)) == '{"DSPLowCutFilterEnabled": "True", '
-        '"DspLowCutFrequency": "10", '
-        '"DspLowCutNumTaps": "0", '
-        '"DspLowCutFilterType": "DCO", '
-        '"DSPHighCutFilterEnabled": "True", '
-        '"DspHighCutFrequency": "9000", '
-        '"DspHighCutNumTaps": "64", '
-        '"DspHighCutFilterType": "FIR", '
-        '"DspDelayCompensation": "Enabled", '
-        '"DspFilterDelay_µs": "984"}'
-    )
-
-
-def test_neuralynx_filtering_recording_extractor():
-
-    ri = NeuralynxRecordingInterface(folder_path=(NLX_PATH / "Cheetah_v5.7.4" / "original_data"))
-    filtering = ri.recording_extractor.get_property("filtering")[0][:]
-    expected_filtering = (
-        '{"DSPLowCutFilterEnabled": "True", '
-        '"DspLowCutFrequency": "10", '
-        '"DspLowCutNumTaps": "0", '
-        '"DspLowCutFilterType": "DCO", '
-        '"DSPHighCutFilterEnabled": "True", '
-        '"DspHighCutFrequency": "9000", '
-        '"DspHighCutNumTaps": "64", '
-        '"DspHighCutFilterType": "FIR", '
-        '"DspDelayCompensation": "Enabled", '
-        '"DspFilterDelay_µs": "984"}'
-    )
-    assert filtering == expected_filtering
+    file_path = NLX_PATH / "Cheetah_v5.7.4" / "original_data"
+    interface = NeuralynxRecordingInterface(file_path)
+    filtering = interface.recording_extractor.get_property("filtering")
+
+    assert interface.recording_extractor.get_num_channels() == len(filtering)
+    assert '"DSPLowCutFilterEnabled": "True"' in filtering[0]
```

Comment:
This should not be a json but we should have a channel property for each of those values. That is, instead of having a single property whose values is a json dictionary with the x, y and z values, we should have a properties x, y and z with values for each channels. This can be done automatically by spikeinterface I think.

Reply:
Yes, I see, so spikeinterface properties correspond to neo array_annotations. I will see how I can make use of this.

Reply:
@h-mayorquin Thanks for the `all_annotations` hint. I am using this now to automatically convert all neo array_annotations to properties. To still keep track of which of those properties are related to filtering I added an annotation that lists all filter related property keys. What do you think of this solution?

Reply:
So where would you add the additional metadata then instead to the NeuralynxInterface? Is there' a way how to add this directly to the NWB metadata without nwb complaining about invalid keys being used?

In general if there would be a standardized way on how to specify filter settings (e.g. specifying multiple filters for each channel with pre-defined keys for each channel already in spikeinterface that would be very useful in my opinion. Possible keys could be filter type ('low'/'high'), specification (e.g. 'butterworth',...), order (int), mode ('analog'/'digital'), ....